documentation/modules/auxiliary/scanner/http/nagios_xi_scanner.md

```diff
@@ -0,0 +1,155 @@
+## Vulnerable Application
+
+The module detects the version of Nagios XI applications and suggests matching exploit modules based on the version number.
+
+The module takes advantage of the `Msf::Exploit::Remote::HTTP::NagiosXi` mixin in order to
+authenticate to the target and obtain the version number, which is only revealed to authenticated users.
+
+When used to target a specific host, the module requires valid credentials for a Nagios XI account.
+These can be provided via `USERNAME` and `PASSWORD` options.
+Alternatively, it is possible to provide a specific Nagios XI version number via the `VERSION` option.
+In that case, the module simply suggests matching exploit modules and does not probe the target(s).
+
+The module currently supports the following exploit modules:
+- exploit/linux/http/nagios_xi_plugins_check_plugin_authenticated_rce (CVE-2019-15949)
+- exploit/linux/http/nagios_xi_plugins_filename_authenticated_rce (CVE-2020-35578)
+- exploit/linux/http/nagios_xi_mibs_authenticated_rce (CVE-2020-5791)
+- exploit/linux/http/nagios_xi_snmptrap_authenticated_rce (CVE-2020-5792)
+
+### Setting up Nagios XI for testing
+
+Vulnerable Nagios XI versions are available [here](https://assets.nagios.com/downloads/nagiosxi/versions.php).
+Detailed installation instructions are available
+[here](https://assets.nagios.com/downloads/nagiosxi/docs/Installing-Nagios-XI-Manually-on-Linux.pdf)
+and an official video tutorial is available [here](https://www.youtube.com/watch?v=fBWA6t6dJ4I).
+
+## Verification Steps
+1. Start msfconsole
+2. Do: `use auxiliary/scanner/http/nagios_xi_scanner`
+3. Do: `set RHOSTS [IP]`
+4. Do: `set USERNAME [username for a valid Nagios XI account]`
+5. Do: `set PASSWORD [password for a valid Nagios XI account]`
+6. Do: `run`
+
+## Options
+### FINISH_INSTALL
+If this is set to `true`, the module will try to finish installing Nagios XI on targets where the installation has not been completed.
+This includes signing the license agreement. The default value is `false`.
+### PASSWORD
+The password for the Nagios XI account to authenticate with.
+### TARGETURI
+The base path to Nagios XI. The default value is `/nagiosxi/`.
+### USERNAME
+The username for the Nagios XI account to authenticate with. The default value is `nagiosadmin`.
+### VERSION
+The Nagios XI version to check against existing exploit modules. If this option is selected,
+the module will not probe the target, so it is not necessary to provide credentials.
+
+## Scenarios
+### Nagios XI 5.6.5 running on CentOS 7
+```
+msf6 auxiliary(scanner/http/nagios_xi_scanner) > show options 
```

Comment:
These scenarios need to show you starting from loading right after you load `msfconsole`. Aka you need to start with the command to load the module, and then the commands to set the various options, before then showing the options and running the module.